enterprise/app/code/code.tsx

```diff
@@ -143,14 +149,16 @@ export default class CodeComponent extends React.Component<Props, State> {
       return;
     }
 
+    let repoResponse = await repoLoadedPromise;
```

Comment:
I think `componentDidMount()` is not supposed to do async stuff, because by the time the promise completes, the component might have already unmounted.

I think it would be slightly better to have a `componentDidUpdate(prevProps, prevState)` that looks at prevState and checks `if (!prevState.repoResponse && this.state.repoResponse) { doAllThisLogicBelow() }` - that would also let you get rid of the promise

Reply:
Great idea - I was able to remove the promise by moving things to `componentDidUpdate`.

I still needed to call this method in `componentDidMount` though because `componentDidUpdate` isn't called on initial render (and if state is stored in localstorage - everything is ready and loaded into state on initial render.)

Definitely feels much cleaner though.